clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -460,15 +462,21 @@ boolean joinGroupIfNeeded(final Timer timer) {
                     exception instanceof IllegalGenerationException ||
                     exception instanceof MemberIdRequiredException)
                     continue;
-                else if (!future.isRetriable())
-                    throw exception;
-
-                timer.sleep(rebalanceConfig.retryBackoffMs);
+                else {
+                    handleFailure(future, timer);
+                }
             }
         }
         return true;
     }
 
+    protected void handleFailure(RequestFuture<?> future, Timer timer) {
+        if (future.isRetriable() || future.exception() instanceof GroupAuthorizationException)
```

Comment:
I'm still confused here, why we would like to sleep when we decide to throw the GroupAuthorizationException afterwards?

Reply:
@abbccdda Thanks for the review, sorry about the delay in following this up. The main purpose of this PR is to add backoff before propagating GroupAuthorizationException so that a client that polls in a loop doesn't end up in a tight poll loop if user doesn't have access to the group. Added comments.